interview-questions.md

```diff
@@ -6,62 +6,64 @@ First, without external resources. Challenge yourself to answer from memory as i
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a javascript library that has a user interface. I would use it because it is very useful. I know I use apps that use react such as facebook and netflix.
 
-  Researched answer:
+  Researched answer: React is a javascript library that creates an interactive UI. Companies like Facebook and Netflix use react because it is a scalable application. Using react can be extremely helpful when it comes down to cutting down development time. For example, react allows developers to create and use components that can be used interchangebly, basically making it reuseable. When something reloads it doesn't refresh the whole page rather it resets that small interaction and isolates it.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: I don't know the method by that name but I am sure that I have used it before. I know class components are normally followed by constructer(props) and super(props).
 
-  Researched answer:
+  Researched answer: After doing a little research I'm surprised to find out that the lifecycle method required for a react class component is render(). It seems that I may have missed it. When render is called you can get either of these items react elements, array and fragments, portals, strings, numbers, boolean, or null. I found out that render should not modify the component state.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: I know that you use JSX during react and you normally use it to "escape" a language. I know one of the difference is that you use a curly bracket as apart of its syntax.
 
-  Researched answer:
+  Researched answer: JSX stands for JavaScriptXml and fun fact it was created as a syntactic sugar for React. When you are writing in JSX you are writing in HTML and javascript together.
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: Yarn is a package manager and holds thousands of node modules. Each one doesn't interfere with one another but when it comes together it works well with each other. I can't recall what file(s) are modified when you run yarn in a react application.
 
-  Researched answer:
+  Researched answer: Yarn is a package manager and it holds a thousand of node modules. The great things about these node modules is that they are isolated. They don't interfere together but when they are put together it turns out great. The files that are modified are yarn.lock and package.json if I am not wrong. 
```

Comment:
Awesome!  Yarn does so much for us behind the scenes and a great place to explore the dependencies we have installed from yarn is in package.json